Xamarin.Forms.Core/Registrar.cs

```diff
@@ -94,37 +94,18 @@ internal TRegistrable GetHandler(Type type, params object[] args)
 
 		public Type GetHandlerType(Type viewType)
 		{
-			Type type;
-			if (LookupHandlerType(viewType, out type))
-				return type;
+			// 1. Do we have this specific type registered already?
+			if (LookupHandlerType(viewType, false, out Type specificTypeRenderer))
+				return specificTypeRenderer;
 
-			// lazy load render-view association with RenderWithAttribute (as opposed to using ExportRenderer)
-			var attribute = viewType.GetTypeInfo().GetCustomAttribute<RenderWithAttribute>();
-			if (attribute == null)
-			{
-				Register(viewType, null); // Cache this result so we don't have to do GetCustomAttribute again
-				return null;
-			}
+			// 2. Do we have a RenderWith for this type or its base types? Register them now.
+			RegisterRenderWithTypes(viewType);
 
-			type = attribute.Type;
-
-			if (type.Name.StartsWith("_", StringComparison.Ordinal))
-			{
-				// TODO: Remove attribute2 once renderer names have been unified across all platforms
-				var attribute2 = type.GetTypeInfo().GetCustomAttribute<RenderWithAttribute>();
-				if (attribute2 != null)
-					type = attribute2.Type;
-
-				if (type.Name.StartsWith("_", StringComparison.Ordinal))
-				{
-					Register(viewType, null); // Cache this result so we don't work through this chain again
-					return null;
-				}
-			}
-
-			Register(viewType, type); // Register this so we don't have to look for the RenderWith Attibute again in the future
-
-			return type;
+			// 3. Do we have a custom renderer for a base type or did we just register an appropriate renderer from RenderWith?
+			if (LookupHandlerType(viewType, true, out Type baseTypeRenderer))
+				return baseTypeRenderer;
```

Comment:
if the renderer is on the base class, shouldn't we cache this value ?

Reply:
Not sure I understand. By this point, all renderers should be cached in `_handlers`. We're just pulling it out of there now.

Reply:
ok